### PR TITLE
Restructure remote parsers

### PR DESCRIPTION
**Pull request by wangkuiyi**
This PR changes the directory structure of `/parser/` after a discussion with @ywskycn .

The new directory structure is as stated here: https://github.com/sql-machine-learning/sqlflow/pull/514/files#diff-b434dfa3690611a8c2a55b8f5feb9847R150

I verified this PR that 

1. I can create a Dockerfile `sqlflow:mvn` from `parser/remote/Dockerfile`, as explained in https://github.com/sql-machine-learning/sqlflow/pull/514/files#diff-bea7959adbf7995e854eceac35967114R9

1. In the container that runs this image, I typed `mvn package`, which builds `parser/remote/hiveql` into `parser/remote/hiveql/targets/hiveql-parser-0.1-SNAPSHOT.jar`, as explained in https://github.com/sql-machine-learning/sqlflow/pull/514/files#diff-bea7959adbf7995e854eceac35967114R22

1. I can run the (incomplete) HiveQL parser as explained in https://github.com/sql-machine-learning/sqlflow/pull/514/files#diff-72b025e26b40cee761d1222e88a98cc8R9.

After the merging of this PR, I will

1. Complete the HiveQL parser by making it a gRPC server.
1. Refactorize the Calcite parser by making it buildable by Maven.
1. Move the Go client of all remote parsers out from `parser/remote/calcite` to `parser/remote/client.go` and make it work with `parser/grpc/**/ParserService.proto`.

### CODE REVIEW

File: parser/README.md [hunk 1]
```diff
@@ -72,18 +72,18 @@ It is notable that MySQL doesn't provide sufficient documentation on how to call
 The above example program calls `sql_engine.Parse` twice before calling `SQLFlow.Parse`.  In practice, because Calcite parser is in Java, SQLFlow is a Go program, to enable SQLFlow calling Calcite parser, we have to wrap Calcite parser up into a gRPC server.  It is time-consuming to make an RPC call, so we pack the two calls to `sql_engine.Parse` into one:
 
 ```protobuf
-message CalciteParserRequest {
+message ParserRequest {
   string query = 1;
 }
 
-message CalciteParserReply {
+message ParserReply {
```
Comment (ywskycn) [inline parser/README.md:79]: Here should use `Response` instead of `Reply`? Also for the following one

File: parser/remote/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Currently, we have two remote parsers, each as a gRPC server in Java:
+
+- hiveql
+- calcite
+
+
+## Development Environment
+
+To make sure that all developers use the same version and configration of development tools like JDK, Maven, and protoc, as well as dependencies like gRPC, we install all of them into a Docker image.
```
Comment (ywskycn) [inline parser/remote/README.md:11]: configration -> configuration

File: parser/remote/hiveql/pom.xml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <source>1.6</source>
```
Comment (ywskycn) [inline parser/remote/hiveql/pom.xml:34]: 1.7 or 1.8? 1.6 seems too old.

**Review by weiguoz — APPROVED** [review]

File: parser/README.md (hunk 1 above)
Comment (wangkuiyi) [reply parser/README.md:79]: Got it. Will fix them in the next PR.

File: parser/remote/README.md (hunk 2 above)
Comment (wangkuiyi) [reply parser/remote/README.md:11]: Got it. Will fix in the next PR. Thank you for being very detail-oriented!

File: parser/remote/hiveql/pom.xml (hunk 3 above)
Comment (wangkuiyi) [reply parser/remote/hiveql/pom.xml:34]: I see. Will change to 1.8 in the next PR!

**Pull request merged**